hera_mc/tests/test_rtp.py

```diff
@@ -382,92 +507,292 @@ def test_errors_rtp_process_record(mcsession, observation, record):
+    }
+    result_obsid_most_recent = get_method(**get_keywords)
+    assert len(result_obsid_most_recent) == 1
+    assert result[1] == result_obsid_most_recent[0]
+
+    result_task_no_times = get_method(
+        task_name=data_obj.task_jobid_columns['task_name'], most_recent=False)
+    assert len(result_task_no_times) == 1
+    assert result_task_no_times[0].isclose(expected)
+
+    result = get_method(
+        starttime=data_obj.task_jobid_columns['start_time']
+        - TimeDelta(2, format='sec'),
+        task_name=data_obj.task_jobid_columns['task_name'],
+        stoptime=data_obj.task_jobid_columns['start_time']
+        + TimeDelta(2 * 60, format='sec'))
+    assert len(result) == 1
+    result = result[0]
+    assert result.isclose(expected)
+
+    result_task_most_recent = get_method(
+        task_name=data_obj.task_jobid_columns['task_name'], most_recent=True)
+    assert len(result_task_most_recent) == 1
+    assert result_task_most_recent[0].isclose(expected)
+
+    new_task_time = (data_obj.task_jobid_columns['start_time']
+                     + TimeDelta(3 * 60, format='sec'))
+
+    new_obsid_time = (data_obj.task_jobid_columns['start_time']
+                      + TimeDelta(3 * 60, format='sec'))
+    new_obsid = utils.calculate_obsid(new_obsid_time)
+    test_session.add_obs(
+        Time(new_obsid_time),
+        Time(new_obsid_time + TimeDelta(10 * 60, format='sec')), new_obsid)
+    obs_result = test_session.get_obs(obsid=new_obsid)
+    assert obs_result[0].obsid == new_obsid
+
+    add_method(
+        new_obsid, data_obj.task_jobid_columns['task_name'], new_task_time,
+        *data_obj.task_jobid_values[3:])
+
+    result = get_method(
+        starttime=data_obj.task_jobid_columns['start_time']
+        - TimeDelta(2, format='sec'),
+        task_name=data_obj.task_jobid_columns['task_name'],
+        stoptime=data_obj.task_jobid_columns['start_time']
+        + TimeDelta(5 * 60, format='sec'))
+    assert len(result) == 2
+
+    result_task_no_times = get_method(
+        task_name=data_obj.task_jobid_columns['task_name'], most_recent=False)
+    assert len(result_task_no_times) == 2
+    assert result_task_no_times == result
+
+    result_task_most_recent = get_method(
+        task_name=data_obj.task_jobid_columns['task_name'], most_recent=True)
+    assert len(result_task_most_recent) == 1
+    assert result_task_most_recent[0].isclose(result[1])
+
+    get_keywords = {
+        "starttime": data_obj.task_jobid_columns['start_time'] - TimeDelta(2, format='sec'),
+        obsid_name: data_obj.task_jobid_columns[obsid_name],
+        "task_name": data_obj.task_jobid_columns['task_name'],
+        "stoptime": data_obj.task_jobid_columns['start_time'] + TimeDelta(5 * 60, format='sec')
+    }
+    result = get_method(**get_keywords)
+    assert len(result) == 1
+    result = result[0]
+    assert result.isclose(expected)
+
+    filename = os.path.join(DATA_PATH, 'test_rtp_task_record_file.csv')
```

Comment:
Do we want to use a temporary directory for this?

Reply:
Yes, thanks. I fixed it in all the similar places too.